integration_test/cloudbuild-integration-test.yaml

```diff
@@ -0,0 +1,137 @@
+  dir: dcm4che
+
+- name: "maven:3.6.0-jdk-8"
+  id: build-store-scu
+  volumes:
+  - name: 'vol1'
+    path: '/root/.m2'
+  args:
+  - mvn
+  - install
+  dir: "dcm4che/dcm4che-tool/dcm4che-tool-storescu"
+
+# run adapter, close when connection established on _CLOSE_ADAPTER_PORT
+- name: 'gradle:5.4.1-jdk8'
+  id: gradle-run-adapter
+  args: [ 'bash', '../integration_test/run_adapter_and_wait.sh', '${_STAGE}', '${_PROJECT}', '${_LOCATION}', '${_DATASET}', '${_STORE_NAME}', '${_ADAPTER_PORT}', '${_CLOSE_ADAPTER_PORT}']
+  dir: import
+  waitFor: ['-']
+
+# delete dicom store if it already exists, ignore exit code
+- name: 'google/cloud-sdk:250.0.0'
+  id: cleanup-dicom-store
+  args:
+  - bash
+  - integration_test/store_delete_ignore_result.sh
+  - ${_LOCATION}
+  - ${_DATASET}
+  - ${_STORE_NAME}
+  waitFor: ['-']
+
+- name: 'google/cloud-sdk:250.0.0'
+  id: create-dicom-store
+  args:
+  - gcloud
+  - alpha
+  - healthcare
+  - dicom-stores
+  - create
+  - ${_STORE_NAME}
+  - "--dataset=${_DATASET}"
+  - "--location=${_LOCATION}"  
+  - "--quiet"
+  waitFor: 
+  - cleanup-dicom-store
+
+
+- name: 'busybox:latest'
+  id: wait-for-adapter
+  args:
+  - ash
+  - integration_test/wait_for_port.sh
+  - ${_ADAPTER_RUN_STEP}
+  - ${_ADAPTER_PORT}
+  waitFor:
+  - build-store-scu
+  - create-dicom-store
+
+- name: "maven:3.6.0-jdk-8"
+  id: run-store-scu
+  volumes:
+  - name: 'vol1'
+    path: '/root/.m2'
+  args:
+  - mvn
+  - 'exec:java'
+  - '-Dexec.mainClass=org.dcm4che3.tool.storescu.StoreSCU'
+  - '-Dexec.args=-c IMPORTADAPTER@step_3:${_ADAPTER_PORT} ../../../integration_test/example.dcm'
+  dir: "dcm4che/dcm4che-tool/dcm4che-tool-storescu"
+  waitFor:
+  - wait-for-adapter
+
+- name: 'busybox:latest'
+  id: close-adapter
+  args:
+  - nc
+  - '-z'
+  - ${_ADAPTER_RUN_STEP}
+  - ${_CLOSE_ADAPTER_PORT}
+  waitFor:
+  - run-store-scu
+
+- name: 'google/cloud-sdk:250.0.0'
+  id: curl-dcm
+  args:
+  - bash
+  - integration_test/curl-dcm.sh
+  - "https://healthcare.googleapis.com/${_STAGE}/projects/${_PROJECT}/locations/${_LOCATION}/datasets/${_DATASET}/dicomStores/${_STORE_NAME}/dicomWeb/studies/1.3.6.1.4.1.5962.1.167.1174395400.11033.0/series/1.3.6.1.4.1.5962.1.176.1174395403.11033.0/instances/1.3.6.1.4.1.5962.1.178.1174395404.11033.0"
+  - integration_test/downloaded.dcm
+  waitFor:
+  - run-store-scu
+
+- name: 'busybox:latest'
+  id: diff-dcm
+  args:
+  - diff
+  - 'integration_test/downloaded.dcm'
+  - 'integration_test/example.dcm'
+  waitFor:
+  - curl-dcm
+
+- name: 'google/cloud-sdk:250.0.0'
+  id: delete-dicom-store
+  args:
+  - gcloud
+  - alpha
+  - healthcare
+  - dicom-stores
+  - delete
+  - ${_STORE_NAME}
+  - "--dataset=${_DATASET}"
+  - "--location=${_LOCATION}"  
+  - "--quiet"
+  waitFor:
+  - diff-dcm
+
+substitutions:
+  _STAGE: v1beta1
+  _PROJECT: healthcare-api-215503
+  _LOCATION: europe-west2
+  _DATASET: example-west
+  _STORE_NAME: IntegrationTestStore
+  _ADAPTER_PORT: '2575'
+  _CLOSE_ADAPTER_PORT: '3000'
```

Comment:
I wasn't aware that the adapter shuts down when a packet is sent to port 3000. Is this functionality from the dcm4che library it's based on?

Reply:
No it isn't. It's from run_adapter_and_wait.sh script. I just run adapter in background (&) and wait on port 3000 via nc to gracefully finish this cloud build step. 
Adapter itself has no shutdown feature, and adding one just for integration test didn't seem right either.